### PR TITLE
De cleanups/20241010/v2

### PR DESCRIPTION
**Pull request by inashivb**
Previous PR: https://github.com/OISF/suricata/pull/11928

Changes since v1:
- rebased on top of latest master
- rename whitelisting setting to priority to reflect correct purpose

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 23087

File: src/detect-engine.c [hunk 1]
```diff
@@ -2915,55 +2915,53 @@ static int DetectEngineCtxLoadConf(DetectEngineCtx *de_ctx)
         }
     }
 
-    /* parse port grouping whitelisting settings */
+    /* parse port grouping priority settings */
 
     const char *ports = NULL;
-    (void)ConfGet("detect.grouping.tcp-whitelist", &ports);
```
Comment (victorjulien) [inline src/detect-engine.c:2921]: we need to handle old configs as well, which means we first read the new `tcp-priorityports` option, but if it doesn't exist, we try `tcp-whitelist`

Btw `tcp-priority-ports` looks better to me.

**Review by victorjulien — CHANGES_REQUESTED** [review]
see inline

File: src/detect-engine.c (hunk 1 above)
Comment (inashivb) [reply src/detect-engine.c:2921]: > we need to handle old configs as well, which means we first read the new `tcp-priorityports` option, but if it doesn't exist, we try `tcp-whitelist`
> 

Indeed. Should be more mindful of backwards compatibility. Thanks a lot! 🙇🏽‍♀️ 

> Btw `tcp-priority-ports` looks better to me.

To me as well. Coming up in the next PR.

**Pull request closed without merging**